app/backend/src/couchers/servicers/conversations.py

```diff
@@ -398,9 +406,9 @@ def CreateGroupChat(self, request, context):
             )
             session.add(your_subscription)
 
-            for recipient in request.recipient_user_ids:
+            for recipient in recipient_user_ids:
```

Comment:
I reckon just move the hidden check into this for loop. Due to db atomicity, it'll roll everything back.

So just add a check here before the friendship check to make sure the user exists and is visible, no need to do the big check above. (Also it should outright fail if there's invalid users, not filter them out).

(I see now there's no explicit check to make sure the user exists, but it'd turn up either in the friend check or when the database foreign key doesn't work)

The messagign stuff needs a refactorign anyway, no need to be super elegant

Reply:
I feel like the check should still come at the beginning. I can loop through all recipients to check they're visible. What if it's only sending a message to one person and that person is invisible, this solution wouldn't catch that.

Reply:
Using is_visible on an individual user was proving impossible, so I just added a length check

Reply:
> What if it's only sending a message to one person and that person is invisible, this solution wouldn't catch that.

Why?

Reply:
Because it comes after and individual message has been sent.

https://github.com/Couchers-org/couchers/blob/4ea4ef3daa6fc658629d59d1d1dfbfe3f018ca08/app/backend/src/couchers/servicers/conversations.py#L363-L409

Reply:
That just makes sure you don't already have another DM with someone when you try to make a new one. It's not sending the individual message

Reply:
Ya, you're right, I misread the indentations. Still, I think a quick length check at the beginning will save a bunch of backend work if one of the recipients is invisible. Plus recipient.is_visible is proving super hard to work with.